### PR TITLE
Unable to index mainnet genesis block

### LINKED ISSUE #1517: Unable to index mainnet genesis block
### Actual behavior

Unable to index mainnet genesis block.
The indexer complains of timeout when requesting the genesis blocks from the connector.

### Expected behavior

The indexing process should proceed without any issues when indexing the mainnet/testnet genesis blocks.

### Steps to reproduce

- Migrate mainnet/testnet state snapshot to generate the new genesis block compatible with Lisk SDK v6
- Start Lisk Core with the new genesis block
- Start Lisk Service against the above Lisk Core instance

### Which version(s) does this affect? (Environment, OS, etc...)

Lisk Service v0.7.0-alpha.7
Lisk Core v4.0.0-alpha.13

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1517

### How was it solved?
- [x] Filter out genesis assets from the genesis block
- [x] Implement methods to retrieve genesis assets for PoS module
  - [x] getPoSGenesisValidators
  - [x] getPoSGenesisStakers
- [x] Updated indexes for `events/eventTopics` table
- [x] Update swagger specification
- [x] Fix issues with invoking the `/blockchain/apps` endpoint when running Lisk Service against sidechains

### How was it tested?
- [x] Local

### CODE REVIEW

File: services/blockchain-indexer/shared/constants.js [hunk 1]
```diff
@@ -84,6 +101,12 @@ const getSystemMetadata = async () => {
 	return systemMetadata;
 };
 
+const getBlockAssetDataSchemaByModule = async (moduleName) => {
+	const metadata = await getSystemMetadata();
+	const module = metadata.modules.find(metaData => metaData.name === moduleName);
+	return module.assets[0].data;
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/constants.js:107]:
```suggestion
	const [asset = {}] = (metadata.modules.find(metaData => metaData.name === moduleName)).assets;
	return asset.data;
```

File: services/blockchain-indexer/shared/utils/genesisBlock.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	Logger,
+} = require('lisk-service-framework');
+
+const { codec } = require('@liskhq/lisk-codec');
+
+const BluebirdPromise = require('bluebird');
+
+const fs = require('fs');
+const json = require('big-json');
+const path = require('path');
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/utils/genesisBlock.js:26]:
```suggestion
const fs = require('fs');
const json = require('big-json');
const path = require('path');
const BluebirdPromise = require('bluebird');

const logger = require('lisk-service-framework').Logger();
const { codec } = require('@liskhq/lisk-codec');
```

File: services/blockchain-indexer/shared/utils/genesisBlock.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	getGenesisBlockUrl,
+	getChainID,
+	getBlockAssetDataSchemaByModule,
+} = require('../constants');
+const { downloadFile, verifyFileChecksum } = require('./download');
+const { parseToJSONCompatObj } = require('./parser');
+
+let genesisBlock = { header: {} };
+
+const logger = Logger();
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/utils/genesisBlock.js:39]:
```suggestion
```

File: services/blockchain-connector/shared/sdk/genesisBlock.js [hunk 4]
```diff
@@ -41,7 +44,8 @@ const getGenesisHeight = async () => {
 const getGenesisBlock = async () => {
 	try {
 		const block = await getGenesisBlockFromFS();
-		return block;
+		// Filter out assets from genesis block and assign empty array
+		return { header: block.header, transactions: block.transactions, assets: [] };
```
Comment (sameersubudhi) [inline services/blockchain-connector/shared/sdk/genesisBlock.js:48]:
```suggestion
		// Filter out assets from genesis block and assign empty array
		return { ...block, assets: [] };
```

File: services/blockchain-connector/shared/sdk/genesisBlock.js [hunk 5]
```diff
@@ -80,9 +84,41 @@ const getGenesisConfig = async () => {
 	}
 };
 
+const getPoSGenesisStakers = async () => {
```
Comment (sameersubudhi) [inline services/blockchain-connector/shared/sdk/genesisBlock.js:87]: This should be implemented in a generic manner where the genesis block is fetched via the getBlock or getGenesisBlock method. Let's discuss this offline.

File: services/blockchain-connector/shared/sdk/genesisBlock.js [hunk 6]
```diff
@@ -80,9 +84,41 @@ const getGenesisConfig = async () => {
...
+		return stakers;
+	} catch (error) {
+		if (error.message.includes(timeoutMessage)) {
+			throw new TimeoutException('Request timed out when calling \'getPoSGenesisStakers\'.');
+		}
+		throw error;
+	}
+};
+
+const getPoSGenesisValidators = async () => {
```
Comment (sameersubudhi) [inline services/blockchain-connector/shared/sdk/genesisBlock.js:102]: Same as above.

File: services/gateway/apis/http-version3/methods/blockAssets.js [hunk 7]
```diff
@@ -41,15 +41,15 @@ module.exports = {
 		const blockAssetSchema = {};
 		blockAssetSchema[this.swaggerApiPath] = { get: {} };
 		blockAssetSchema[this.swaggerApiPath].get.tags = this.tags;
-		blockAssetSchema[this.swaggerApiPath].get.summary = 'Requests block assets data';
+		blockAssetSchema[this.swaggerApiPath].get.summary = 'Requests block assets data.';
 		blockAssetSchema[this.swaggerApiPath].get.description = getSwaggerDescription({
 			rpcMethod: this.rpcMethod,
-			description: 'Returns block assets data',
+			description: 'Returns block assets data. Block assets are always returned empty for genesis height.',
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/methods/blockAssets.js:47]:
```suggestion
			description: 'Returns block assets data. Assets are always returned empty for the genesis height.',
```

File: services/gateway/apis/http-version3/swagger/definitions/blocks.json [hunk 8]
```diff
@@ -210,16 +209,27 @@
...
+                        "data": {
+                            "type": "object",
+                            "properties": {
+                                "seedReveal": {
+                                    "type": "string",
+                                    "example": "c555338417025a5b2b9278304f8a96be"
+                                }
+                            }
+                        }
+                    }
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/blocks.json:231]:
```suggestion
                    },
                    "required": [
                        "module",
                        "data"
                    ],
```

File: services/gateway/apis/http-version3/swagger/definitions/blocks.json [hunk 9]
```diff
@@ -210,16 +209,27 @@
...
+                            "description": "Name of the blockchain module associated with the assets."
+                        },
+                        "data": {
+                            "type": "object",
+                            "properties": {
+                                "seedReveal": {
+                                    "type": "string",
+                                    "example": "c555338417025a5b2b9278304f8a96be"
+                                }
+                            }
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/blocks.json:229]:
```suggestion
                            },
                            "description": "The data object schema varies by module. The exact schema can be found from the `/schemas`endpoint."
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-connector/shared/sdk/genesisBlock.js [hunk 10]
```diff
@@ -38,10 +39,14 @@ const getGenesisHeight = async () => {
 	return genesisHeight;
 };
 
-const getGenesisBlock = async () => {
+const getGenesisBlock = async (includeGenesisAssets) => {
```
Comment (sameersubudhi) [inline services/blockchain-connector/shared/sdk/genesisBlock.js:42]:
```suggestion
const getGenesisBlock = async (isIncludeAssets = false) => {
```

File: services/blockchain-connector/shared/sdk/genesisBlock.js [hunk 11]
```diff
@@ -38,10 +39,14 @@ const getGenesisHeight = async () => {
...
 	try {
 		const block = await getGenesisBlockFromFS();
-		return block;
+		// Filter out assets from genesis block and assign empty array
+		return {
+			...block,
+			assets: includeGenesisAssets ? block.assets : [],
```
Comment (sameersubudhi) [inline services/blockchain-connector/shared/sdk/genesisBlock.js:48]:
```suggestion
			assets: isIncludeAssets ? block.assets : [],
```

File: services/gateway/apis/http-version3/swagger/definitions/blocks.json [hunk 12]
```diff
@@ -210,16 +209,32 @@
...
+                        },
+                        "data": {
+                            "type": "object",
+                            "properties": {
+                                "seedReveal": {
+                                    "type": "string",
+                                    "example": "c555338417025a5b2b9278304f8a96be"
+                                }
+                            },
+                            "description": "The data object schema varies by module. The exact schema can be found from the `/schemas`endpoint."
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/blocks.json:234]:
```suggestion
                            "description": "The data object schema varies by module. The exact schema can be found from the `/schemas` endpoint."
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Review by nagdahimanshu — APPROVED** [review]

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Pull request merged**